### ISSUE TITLE
Android Studio Arctic Fox // 2020.3.1 Build Error

### ISSUE THREAD

**Issue by SongHuaixu** [flutter/flutter#87249]
Old Version Java Path
`/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home/bin/java`

New Version Java Path 
`/Applications/Android\ Studio.app/Contents/jre/Contents/Home/bin/java`

```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.4 20F71 darwin-x64, locale
    zh-Hans-CN)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ Cannot execute /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java to determine the
      version

```

**Comment by andhie** [comment]
As a workaround, you can install JDK8 and set a JAVA_HOME path instead of depending on Android Studio internal JDK.
Based on the comment in [android_sdk.dart#L409](https://github.com/flutter/flutter/blob/76db0d7428f73fcf2d0b4de8fb493bb4ca056380/packages/flutter_tools/lib/src/android/android_sdk.dart#L409)
```
  /// First try Java bundled with Android Studio, then sniff JAVA_HOME, then fallback to PATH.
  static String? findJavaBinary({
  ```

For macOS, you can test it using this command `/usr/libexec/java_home -v 1.8` if it returns something. [android_sdk.dart#L431](https://github.com/flutter/flutter/blob/76db0d7428f73fcf2d0b4de8fb493bb4ca056380/packages/flutter_tools/lib/src/android/android_sdk.dart#L431)

**Comment by SongHuaixu** [comment]
> As a workaround, you can install JDK8 and set a JAVA_HOME path instead of depending on Android Studio internal JDK.
> Based on the comment in [android_sdk.dart#L409](https://github.com/flutter/flutter/blob/76db0d7428f73fcf2d0b4de8fb493bb4ca056380/packages/flutter_tools/lib/src/android/android_sdk.dart#L409)
> 
> ```
>   /// First try Java bundled with Android Studio, then sniff JAVA_HOME, then fallback to PATH.
>   static String? findJavaBinary({
> ```
> 
> For macOS, you can test it using this command `/usr/libexec/java_home -v 1.8` if it returns something. [android_sdk.dart#L431](https://github.com/flutter/flutter/blob/76db0d7428f73fcf2d0b4de8fb493bb4ca056380/packages/flutter_tools/lib/src/android/android_sdk.dart#L431)

My `JAVA_HOME` is correct, I have changed it to `/Applications/Android\ Studio.app/Contents/jre/Contents/Home/bin/java`, but it still gives me an error. Does it have to be Java 1.8?

**Comment by andhie** [comment]
I have no problem in building my project, so i'm not sure if it has to be Java 8. 
My `JAVA_HOME` is set to `/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home`.

and when i do this in terminal
```
$ /usr/libexec/java_home -v 1.8
/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
```

Hope it helps

**Comment by SongHuaixu** [comment]
> I have no problem in building my project, so i'm not sure if it has to be Java 8.
> My `JAVA_HOME` is set to `/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home`.
> 
> and when i do this in terminal
> 
> ```
> $ /usr/libexec/java_home -v 1.8
> /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
> ```
> 
> Hope it helps

I don't have Java 1.8 installed and I've been using Android Studio internal JDK. So I'm having this problem.

Thank you very much for your help!

**Comment by TahaTesser** [comment]
Hi @SongHuaixu 
new java path in Arctic fox is a known issue https://github.com/flutter/flutter/issues/85602 and fix is blocked by [this](https://github.com/flutter/flutter/issues/85602#issuecomment-886992820) 

However, you should be able use Java from [this](https://adoptopenjdk.net/) 

Mine is set to his 
```
export JAVA_HOME=/Users/tahatesser/Code/jdk/Contents/Home
export PATH=$PATH:$JAVA_HOME/bin
```

Closing as duplicate of https://github.com/flutter/flutter/issues/86238 https://github.com/flutter/flutter/issues/85602

**Comment by negiot** [comment]
flutter doctor -v
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.1 20G80 darwin-x64, locale it-IT)
    • Flutter version 2.2.3 at /Users/negiot/development/flutter
    • Framework revision f4abaa0735 (4 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/negiot/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/negiot/Library/Android/sdk
    • Java binary at: /usr/bin/java
    ✗ Could not determine java version

[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 92.0.4515.107

! Doctor found issues in 3 categories.

**Comment by TahaTesser** [comment]
@negiot 
Android Studio java issue is fixed on master channel
